doc/deployment_guide.md

```diff
@@ -0,0 +1,797 @@
+
+## What is an MDIP Node?
+
+MDIP **nodes** provide the services required to interact with other MDIP nodes in a fully peer-to-peer trustless fashion; the network of nodes being able to reach bysantine consensus on sequences of user-generated MDIP DID operations. An MDIP node is composed of multiple subsystems, each providing essential services for protocol operations.
+
+Services provided by a **Full** or **"Trustless"** node includes:
+
+1. MDIP Gatekeeper API for DID operations.
+1. Local database and storage for DIDs and MDIP operations retention.
+1. Data Swarming Protocol for DIDs and MDIP operations distribution.
+1. One or more registry mediator systems for DIDs and MDIP operations registration.
+1. One or mode blockchain networks for decentralized consensus on order of DID operations.
+
+### Node Communication
+
+At the end of this installation guide, a full MDIP node will expose limited and selected services over HTTP/SSL on port 443. All other ports can be firewalled. MDIP nodes communicate with each other during the documents Distribution and Registration processes.
+
+- Document Distribution is *fast* and performed using the [Hyperswarm Protocol](https://github.com/holepunchto/hyperswarm). When a Keymaster Client presents a valid MDIP operation to the Gatekeeper API, the Gatekeeper communicates the operation to other Gatekeepers over a public Hyperswarm channel. Distribution of a DID is near-real-time.
+  
+- Document Registration is *immutable* and performed using the MDIP Satoshi Mediator and associated blockchain software. DIDs are registered on a ledger to evidence an order of operation across all MDIP nodes. Registration of a DID is a parallel asynchronous process that occurs at "block" speed.
+
+### MDIP Operations
+
+A system is said to be an MDIP Client if it implements MDIP Keymaster functionality. The MDIP Keymaster library and tools made available in this repository implement functionality to create valid MDIP operations. MDIP Keymaster is available as a REST OpenAPI interface, Command Line Interface, Web User Interface, NodeJS and Python SDKs, npm library, with more additions planned.
+
+Each Keymaster Client manages a **private seed and key** used to sign or decrypt MDIP operations for a specific wallet.
+
+MDIP nodes that need to issue MDIP Challenges, verify Responses, issue Credentials, etc will need an MDIP Keymaster to manage the server's MDIP Wallet containing the keys to the server's identity on the network.
+
+MDIP operators can interact with their server's Keymaster in multiple ways:
+
+- use the `./kc` CLI tool or
+- use the Keymaster WebUI at `http://localhost:4226` or
+- use the Keymaster REST OpenAPI at `http://localhost:4226/api/v1/`
+
+The MDIP [Integration Guide](integration_guide.md) focuses on Keymaster deployments, Keymaster OpenAPI interface, and integration of MDIP wallet functionality with 3rd party applications.
```

Comment:
@Flaxscrip what integration guide is this pointing to?

Reply:
None yet. When I wrote this, I didn't think the deployment guide would be this long. I assumed I'd have the integration guide done within days.

Reply:
Should we plan on such a guide existing before this is released, or edit/remove this line?

Reply:
Created the integrations guide document and updated link in deployment guide.
https://github.com/KeychainMDIP/kc/commit/66b2b05a255d789a2ffdc4e4bb99597a3058c767